### PR TITLE
Add subs

### PR DESCRIPTION
**Pull request by stw**
I find myself adding a subs filter all the time, thought it might be worth adding to Selmer. 

Thanks,
Steve

### CODE REVIEW

**Comment by yogthos** [comment]
thanks, looks great

**Comment by yogthos** [comment]
and just pushed out a new version to Clojars with the update

**Comment by stw** [comment]
Awesome, thank you! And thank you for all your open source work!!

**Comment by yogthos** [comment]
Thanks, glad it's useful. :)

File: README.md [hunk 1]
```diff
@@ -523,6 +524,10 @@ Like the clojure function `str`. So you can do crazy stuff like:
 `(render "{{people|length-is:2|str|join:\"-\"}} lad{{people|pluralize:\"y\":\"ies\"}}" {:people [1 2]})` => `"t-r-u-e ladies"`
 Without raising an exception.
 
+#### subs
+Like the clojure function `subs`.
+`(render "{{s|subs:0:4:\" ..."}}", {:s "Foo bar baz"}) => `"Foo ..."`
```
Comment (bitti) [inline README.md]: 3rd `"` should be escaped

File: src/selmer/filters.clj [hunk 2]
```diff
@@ -77,6 +77,16 @@ map. The rest of the arguments are optional and are always strings."
             :str
             str
 
+            ;;; Like the subs, only adds rest if s is modified
```
Comment (bitti) [inline src/selmer/filters.clj]: The functionality for `rest` is not in `clojure.core/subs`. Shouldn't that be documented?

**Comment by bitti** [comment]
For the usecase implied by the example (that is: shorten too long strings with an ellipsis), this filter is not very useful. For one thing it throws an exception when the string is shorter than the second argument, so it can only be used if the string is known to be at least as long. Another problem is that strings end up longer when they are just 1 character (or in fact just any number of characters equal or less than the `rest` size) too long. E.g. `(render "{{s|subs:0:10:\" ...\"}}", {:s "Foo bar baz"})` yields `Foo bar ba ...` which doesn't make sense. 

I guess no one is hurt by keeping this filter but I would like something along the lines of the `abbreviate` methods in apache commons https://commons.apache.org/proper/commons-lang/apidocs/org/apache/commons/lang3/StringUtils.html#abbreviate-java.lang.String-int-. The undocumented `rest` parameter for this filter could then be removed to make it consistent with `clojure.core/subs` and to avoid confusion about the applicable usecase.

**Comment by yogthos** [comment]
That sounds reasonable to me, and I'd be ok with a pr to improve the behavior if you'd be interested.

**Comment by yogthos** [comment]
I guess the tricky part is accounting for different use cases. Sounds like the filter should take optional hints to decide how the text will be abbreviated. I'd be happy to take a PR for this.

**Pull request merged**